app/v2/controllers/V2DeparturesController.scala

```diff
@@ -159,20 +161,42 @@ class V2DeparturesControllerImpl @Inject() (
         .asPresentation
     } yield declarationResult
 
-  def getMessage(departureId: MovementId, messageId: MessageId): Action[AnyContent] =
+  def getMessage(departureId: MovementId, messageId: MessageId, convertBodyToJson: Boolean): Action[AnyContent] = {
+
+    def checkAndConvertBodyToJson(messageSummary: MessageSummary)(implicit hc: HeaderCarrier): EitherT[Future, PresentationError, MessageSummary] =
+      if (!convertBodyToJson || messageSummary.body.isEmpty) EitherT.rightT(messageSummary)
+      else {
+        for {
+          jsonSource <- conversionService.jsonToXml(messageSummary.messageType, Source.single(ByteString(messageSummary.body.get))).asPresentation
+          jsonBody <- EitherT {
+            jsonSource
+              .fold("")(
+                (acc, byteStr) => acc + byteStr.utf8String
+              )
```

Comment:
It'll be more effcient to concatenate `ByteStrings`, and we could use reduce, so we could do

```
.reduce(_ ++ _).map(_.utf8String)
```

instead.